jsonrpc/helper.go

```diff
@@ -0,0 +1,227 @@
+package jsonrpc
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+var (
+	ErrHeaderNotFound           = errors.New("header not found")
+	ErrPendingBlockNumber       = errors.New("fetching the pending header is not supported")
+	ErrNegativeBlockNumber      = errors.New("invalid argument 0: block number larger than int64")
+	ErrFailedFetchGenesis       = errors.New("error fetching genesis block header")
+	ErrNoDataInContractCreation = errors.New("contract creation without data provided")
+)
+
+type latestHeaderGetter interface {
+	Header() *types.Header
+}
+
+// GetNumericBlockNumber returns block number based on current state or specified number
+func GetNumericBlockNumber(number BlockNumber, store latestHeaderGetter) (uint64, error) {
+	switch number {
+	case LatestBlockNumber:
+		return store.Header().Number, nil
+
+	case EarliestBlockNumber:
+		return 0, nil
+
+	case PendingBlockNumber:
+		return 0, ErrPendingBlockNumber
+
+	default:
+		if number < 0 {
+			return 0, ErrNegativeBlockNumber
+		}
+
+		return uint64(number), nil
+	}
+}
+
+type headerGetter interface {
+	Header() *types.Header
+	GetHeaderByNumber(uint64) (*types.Header, bool)
+}
+
+// GetBlockHeader returns a header using the provided number
+func GetBlockHeader(number BlockNumber, store headerGetter) (*types.Header, error) {
+	switch number {
+	case LatestBlockNumber:
+		return store.Header(), nil
+
+	case EarliestBlockNumber:
+		header, ok := store.GetHeaderByNumber(uint64(0))
+		if !ok {
+			return nil, ErrFailedFetchGenesis
+		}
+
+		return header, nil
+
+	case PendingBlockNumber:
+		return nil, ErrPendingBlockNumber
+
+	default:
+		// Convert the block number from hex to uint64
+		header, ok := store.GetHeaderByNumber(uint64(number))
+		if !ok {
+			return nil, fmt.Errorf("error fetching block number %d header", uint64(number))
+		}
+
+		return header, nil
+	}
+}
```

Comment:
Small suggestion in order to keep code compact, although double check if my proposal is correct.

The only downside I see currently is that for the `EarliestBlockNumber` we would access to the `CurrentHeader` twice :slightly_frowning_face: 

```suggestion
func GetBlockHeader(number BlockNumber, store headerGetter) (*types.Header, error) {
	num, err := GetNumericBlockNumber(number, store)
	if err != nil {
		return nil, fmt.Errorf("error resolving block number %d: %w", number, err)
	}

	header, ok := store.GetHeaderByNumber(num)
	if !ok {
		return nil, fmt.Errorf("error fetching block number %d header", uint64(number))
	}

	return header, nil
}
```

Reply:
it's nice suggestion but more DB accesses happen sometimes and need to fix mock

Reply:
for example, you can get latest header from DB directly but this new function fetch data twice (getting header number and header by number)

Reply:
Yep, aware of it. Ok we are going to leave it as it is currently. :slightly_smiling_face: